extensions/wikia/ArticleComments/src/Hooks/UserCan/Check/CheckFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace Extensions\Wikia\ArticleComments\Hooks\UserCan\Check;
+
+use Extensions\Wikia\ArticleComments\Hooks\UserCan\Action;
+use Extensions\Wikia\ArticleComments\Hooks\UserCan\DependencyFactory;
+
+class CheckFactory {
+	/** @var DependencyFactory $dependencyFactory */
+	private $dependencyFactory;
+
+	public function __construct( DependencyFactory $factory ) {
+		$this->dependencyFactory = $factory;
+	}
+
+	/**
+	 * Given a MediaWiki action, return the corresponding check class, or null if there is none.
+	 *
+	 * @see Action
+	 * @param string $action
+	 * @return Check|null
+	 */
+	public function newActionCheck( string $action ) {
+		switch ( $action ) {
+			case Action::CREATE:
+				return new CreateActionCheck( $this->dependencyFactory );
+				break;
+			case Action::EDIT:
+				return new EditActionCheck( $this->dependencyFactory );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the code after this "return". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1763)